main.c

```diff
@@ -643,6 +644,18 @@ int second_stage_boot(libusb_device_handle *usb_device)
 
 FILE * check_file(const char * dir, const char *fname, int use_fmem)
 {
+	// Check if directory exists
+	DIR* directory = opendir(dir);
+	if (directory)
+	{
+		closedir(directory);
+	}
+	else
+	{
+		fprintf(stderr, "Directory '%s' does not exists\n", dir);
```

Comment:
Minor comment, I think it should be "does not exist" rather than "not exists" but looks good otherwise